osbuild-composer.spec

```diff
@@ -124,6 +124,14 @@ export GOPATH=$GO_BUILD_PATH:%{gopath}
 export GOFLAGS=-mod=vendor
 %endif
 
+# Set the commit hash so that composer can report what source version
+# was used to build it. This has to be set explicitly when calling rpmbuild,
+# this script will not attempt to automatically discover it.
+%if %{?commit:1}0
+export LDFLAGS="${LDFLAGS} -X 'github.com/osbuild/osbuild-composer/internal/common.GitRev=%{commit}'"
+%endif
+export LDFLAGS="${LDFLAGS} -X 'github.com/osbuild/osbuild-composer/internal/common.RpmVersion=%{name}-%{version}-%{release}'"
```

Comment:
Should this be the full NEVRA?

Reply:
Definitely epoch, but I'm not sure about ARCH. Is it not available as an RPM variable. Though it might be useful if we had issues with depsolving on different arch (which we have already had IIRC).

Reply:
Not sure if available everywhere but it is at least on Fedora 34: `%{_arch}`

Reply:
Let's try it in our CI ;-)